cli/ingress-controller/main.go

```diff
@@ -453,8 +455,17 @@ func main() {
 
 	store := store.New(cacheStores, cliConfig.IngressClass, cliConfig.ProcessClasslessIngressV1Beta1,
 		cliConfig.ProcessClasslessIngressV1, cliConfig.ProcessClasslessKongConsumer, log.WithField("component", "store"))
+
+	var dumpDir string
+	if cliConfig.DumpConfig > 0 {
+		var err error
+		dumpDir, err = ioutil.TempDir("", "controller")
+		if err != nil {
+			log.Fatalf("failed to create a dump directory: %v", err)
+		}
+	}
 	kong, err := controller.NewKongController(ctx, &controllerConfig, updateChannel,
-		store)
+		store, dumpDir)
```

Comment:
`NewKongController` takes a lot of arguments, which are grouped in `controllerConfig`. Why add a new positional argument instead of extending `controllerConfig`?